# PARAMS FOR ALL TASKS/TASK UTILITIES/PLANNERS 

# TAKEOFF TASK
# Delay between arming and taking off
delay_before_takeoff    : 2.0
# Velocity with which to take off with
takeoff_velocity        : 0.3
# Height at which to consider takeoff finished in meters
# Should be higher than min_maneuver_height
takeoff_complete_height: 0.7
# Altitude when we transition from acro to angle mode
takeoff_angle_mode_height: 0.2

# LAND TASK
# Velocity with which to land with
land_velocity           : -0.2
# Height below which landing is considered finish
land_height_tolerance   : 0.15

# TRANSLATE-STOP PLANNER
# Max acceleration in the xy plane in m/s^2
max_translation_acceleration: 1.0
# Max acceleration in the xy plane in m/s^2
desired_translation_acceleration: 1.0
# Max translation speed in the xy plane in m/s
max_translation_speed: 3.0
# maximum z velocity 
max_z_velocity: .3
# Tolerance for a position hold in meters
translation_position_hold_tolerance: 0.05
# Translation is considered finished if x,y,z within this range of target
translation_xyz_tolerance : 0.3

# TRACK, BLOCK, AND HIT ROOMBA
# p-term for X direction when tracking roombas 
k_term_tracking_x: .45
# p-term for Y direction when tracking roombas 
k_term_tracking_y: .45

# TRACK ROOMBA
# height at which roomba tracking takes place
track_roomba_height: 1.5
# maximum radial distance from roomba
max_roomba_dist: 1.1

# HIT AND BLOCK ROOMBA
# maximum radial distance from roomba
hit_roomba_max_start_dist: 2.0
# descent velocity when hitting/blocking a roomba 
hit_descent_velocity: -0.2

# BLOCK ROOMBA
# roomba diameter in meters
roomba_diameter: .33
# drone landing gear width and length
drone_landing_gear_width: .775

# HEIGHT HOLDER
# p-term for height holder helper for Z height
p_term_height_hold_z: .90
# maximum z error for heigth holder
max_z_error: 1.0

# HOLD POSITION TASK
# p-term for x 
k_position_x: 1.25
# p-term for y
k_position_y: 1.25
# p-term for z
k_position_z: .90
# maximum allowed radial distance from target point
max_holding_range: 1.0

# HEIGHT RECOVERY TASK
# height offset above min man height, in meters
recover_height_offset: 0.25
